src/client/Microsoft.Identity.Client/AuthScheme/PoP/PoPAuthenticationScheme.cs

```diff
@@ -66,24 +73,24 @@ public string FormatAccessToken(MsalAccessTokenCacheItem msalAccessTokenCacheIte
                 { JsonWebTokenConstants.ReservedHeaderParameters.Type, PoPRequestParameters.PoPTokenType}
             };
 
-            JToken popAssertion = JToken.Parse(_popCryptoProvider.CannonicalPublicKeyJwk);
+            JToken popAssertion = JToken.Parse(_popAuthenticationConfiguration.PopCryptoProvider.CannonicalPublicKeyJwk);
 
             var payload = new JObject(
                 new JProperty(PoPClaimTypes.At, msalAccessTokenCacheItem.Secret),
                 new JProperty(PoPClaimTypes.Ts, (long)(DateTime.UtcNow - s_jwtBaselineTime).TotalSeconds ),
-                new JProperty(PoPClaimTypes.HttpMethod, _httpRequestMessage.Method.ToString()),
-                new JProperty(PoPClaimTypes.Host, _httpRequestMessage.RequestUri.Host),
-                new JProperty(PoPClaimTypes.Path, _httpRequestMessage.RequestUri.AbsolutePath),
+                new JProperty(PoPClaimTypes.HttpMethod, _popAuthenticationConfiguration.PopHttpMethod.ToString()),
```

Comment:
null ref here, since HttpMethod is optional.
if HttpMethod is null, token should not be bound to it.